src/mavsdk/plugins/telemetry/telemetry_impl.cpp

```diff
@@ -1176,6 +1181,42 @@ void TelemetryImpl::process_battery_status(const mavlink_message_t& message)
     }
 }
 
+void TelemetryImpl::process_battery_status_v2(const mavlink_message_t& message)
+{
+    mavlink_battery_status_v2_t bat_status;
+    mavlink_msg_battery_status_v2_decode(&message, &bat_status);
+
+    _has_bat_status = true;
+
+    Telemetry::Battery new_battery;
+    new_battery.id = bat_status.id;
+    new_battery.voltage_v = bat_status.voltage; 
+    //Is it correct to set this to NaN for UINT32_MAX? HOw do you specify = NaN?
+
+    // FIXME: it is strange calling it percent when the range goes from 0 to 1.
+    // Can we fix this? What about if value not provided?
+    new_battery.remaining_percent = bat_status.percent_remaining; // to test * 1e-2f;
+
+    //bat_status also has:
```

Comment:
Other fields are shown below. I think it worth supplying everything. Thoughts? 

My understanding is I'd update the proto and rebuild the file for that.

Reply:
The current seems like a good indicator, and status flags, and temperature is also important.
I would want to ignore the mAh fields as they are somewhat battery specific/internal, at least for smart batteries.

Reply:
Really depends on what you expect your consumers to be able to do with the info. Right now, we're saying that the only thing of interest to a user is the % battery remaining. [technically we're also giving them the voltage, but this is of no use without context provided by SMART_BATTERY_INFO].

With the update I think you're saying that we might also give them current. But what does that actually tell them - without knowing capacity and consumption, you can't do anything with it. So we really need to supply "all or nothing". 

The same thing is true of temperature. If we just want to know that we're over temp we might as well just pass through the status bits, which tell you that the system thinks the battery has a high/low temperature. I guess passing through the actual temperature is useful if we want someone to be able to do something to control it - i.e. know it is not just faulting, but is really about to catch fire.

So I think I need more direction about how you see the users and exactly what fields you would be happy with.

One option might be just a little more - status bits and temperature.

If we're not providing all the extra info then we probably also don't need https://mavlink.io/en/messages/common.html#SMART_BATTERY_INFO, except perhaps for "function".

Reply:
I would be ok with just adding the failure/status flags. I wouldn't necessarily add everything until someone needs/wants it.

Reply:
Hi,
Sounds like a nice improvement, I was thinking about an extended parameter set too.

What I would like to have, as an end user, is how much capacity of my battery I have used.

Also, current is another one. Maybe not as end user parameter but as power consumption, since it's a hint how healthy a setup is.

cheers

Reply:
@julianoes OK, I'll add the flags. Request here, but 

@reedev That said, I stick by my all or nothing :-)

1. Current isn't actually a great measure (anecdotally) of health. It's a good measure of dynamic power consumption, so we would send that for systems that want to do their own estimates of system load/consumption.
2. "How much capacity of my battery I have used" is _problematic_. We can and do supply `current_consumed` but that is the amount consumed _since the battery was turned on_. If you started with a full battery it will be the same thing, but not otherwise.
   - a power module will make the assumption the battery was full, which is how it calculates the remaining charge and 
3. `current_remaining` is the capacity remaining in mAH. It should only sent by a smart battery, which is capable of working out the charge state of the battery reliably. A power module should not send this - a guess based on  `current_consumed` could be widely off.
  - @julianoes I'm wondering if we'd be better off setting a flag to indicate that current_remaining is an estimate based on current consumed - rather than this kind of assumtion?

Reply:
OK,

> Current consumed, and total current is something that needs to be kept internal to a smart battery, in my opinion, because otherwise it will - more often than not - be wrong.

A smart battery can report consumption (since full) and remaining battery capacity reliably -they will not be wrong, even after many power cycles and for aged batteries.
Power monitors only measure consumption since power on, and estimate power remaining based on assumption of a full battery at power on. 

We are going to allow a consumer to determine which case they have using a new flag. If the flag is set (by a smart battery) the values are assumed to be accurate and consumed is "relative to a full battery". If not set (default, or by a power module) only the current consumed can be trusted.

So there is the option to supply this information only if it is known to be accurate, or to supply it and state that it might not be accurate. 

Just FYI!

Reply:
When you have a smart battery you don't need consumption since full and capacity, that's why a smart battery gives you percentage.

Reply:
I agree with only exposing the current. I am not sure how QGC handles the 'current consumed' but that's not something for an FC to keep track with. Current is a raw value and works well. Until now I was peeking the mavlink message itself to calculate the power consumption and I find it very valuable.

I am looking forward to a proper smart battery. I've once purchased a BatMon but never got into it seriously. Are their changes in the PX4?

Reply:
The battery message updates are "in discussion". There are no changes to PX4 yet, but this is part of what might end up happening.